concrete-commons/Cargo.toml

```diff
@@ -17,4 +17,4 @@ crate-type = ["cdylib", "rlib"]
 bench = false
 
 [dependencies]
-serde = { version = "1.0", features = ["derive"]}
\ No newline at end of file
+serde = { version = "1.0", features = ["derive"], optional = true }
```

Comment:
I discovered something while reviewing your PR: you can use depencies names (like `serde` as feature gate). 

I knew you could use them to create feature-dependencies like you did in the `Cargo.toml`:
```toml
serde_serialization = ["serde", "concrete-commons/serde"]
```
But I never thought about writing something like `cargo build --features=serde`. 

At the very least we should have a `serde_serialize` feature in `concrete-commons/Cargo.toml`, because I think some users may miss this otherwise.